pytest_pootle/fixtures/pootle_fs/plugin.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+
+import pytest
+
+
+_plugin_fetch_base = {
+    'conflict': 1,
+    'conflict_untracked': 1,
+    'fs_ahead': 1,
+    'fs_removed': 1,
+    'fs_untracked': 2,
+    'pootle_ahead': 1,
+    'pootle_removed': 1,
+    'pootle_untracked': 1}
+
+FETCH_PATHS = [
+    ("all", None, None),
+    ("pp_short", "/en/tutorial", None),
+    ("pp_no_slash", "en/tutorial/*", None),
+    ("pp_all", "/*", None),
+    ("pp_langs", "/[ez][nu]/project0/*", None),
+    ("pp_pofile", "/en/project0/subdir1/example10.po", None),
+    ("pp_podir", "/en/project0/*example10*", None),
+    ("pp_conflict", "/langage0/project0/subdir3/subsubdir/example4.po", None),
+    ("pp_conflict_untracked", "/subdir3/subsubdir/example5.po", None),
+    ("fs_short", None, "/gnu_style_named_folders"),
+    ("fs_no_slash", None, "en/project0/*"),
+    ("fs_all", None, "/*"),
+    ("fs_gnu", None, "/*gnu*/*"),
+    ("fs_en", None, "/gnu_style/*en*"),
+    ("fs_en.po", None, "/gnu_style/po/en.po")]
+
+
+def _generate_fetch_fixtures():
+    fetch = OrderedDict()
+    for test, pootle_path, fs_path in FETCH_PATHS:
+        fetch[test] = {
+            "pootle_path": pootle_path,
+            "fs_path": fs_path}
+    return fetch
+
+
+def _generate_add_fixtures():
+    add = OrderedDict()
+    for test, pootle_path, fs_path in FETCH_PATHS:
+        add[test] = {
+            "pootle_path": pootle_path,
+            "fs_path": fs_path}
+    return add
+
+
+def _generate_rm_fixtures():
+    rm = OrderedDict()
+    for test, pootle_path, fs_path in FETCH_PATHS:
+        rm[test] = {
+            "pootle_path": pootle_path,
+            "fs_path": fs_path}
+    return rm
```

Comment:
What is the difference with `_generate_add_fixtures()`? This is basically repeated code.